src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java

```diff
@@ -969,8 +990,22 @@ private SortingCollection<GenericRecord> createSortedVetCollectionFromExtractTab
         // We need to look upstream MAX_DELETION_SIZE bases in case there is a deletion that begins before
         // the requested range, but spans into our processing range.  We don't use a "length" or end position
         // because it would break the clustering indexing
+
+        // We want to look upstream... but don't want to go past the beginning of a chromosome.  Check for underflow by
+        // explicitly checking the chromosome by decoding the adjustment to locations that we make for chromosome.  If
+        // we accidentally switch chromosomes, just start at the beginning of the chromosome that we started on.
+        long adjustedStartingLocation = (minLocation - IngestConstants.MAX_DELETION_SIZE + 1);
+
+        long startingChromosome = minLocation / SchemaUtils.chromAdjustment;
+        long adjustedStartingChromosome = adjustedStartingLocation / SchemaUtils.chromAdjustment;
+
+        if (startingChromosome != adjustedStartingChromosome) {
+            // This is just the start of the chromosome
+            adjustedStartingLocation = startingChromosome * SchemaUtils.chromAdjustment;
+        }
+
```

Comment:
This block appears to be a copypasta of the block above, can this be DRYed out into a function?

Reply:
maybe... similar logic appears in 3 places, though.  Two of them are the same, and one is just a bit different.  For only two instances and knowing that I'd have to leave similar-looking logic on a third, it didn't feel QUITE enough to justify factoring it out into a function.  I'm open to reconsider that decision, though

Reply:
I simplified this logic some using Math.max, as suggested for the other block.  But until it's used in more than 2 places, I don't think I'm going to factor it out yet